sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,560 @@
+        {
+          formName: "studentDependantsAppealPartTime",
+          formData: {
+            hasDependents: "yes",
+            dependants: [
+              {
+                fullName: "Mike HOZZENFEUGEN",
+                dateOfBirth: "2023-01-01",
+                attendingPostSecondarySchool: "no",
+                declaredOnTaxes: "no",
+                validDependent: 1,
+                isavalidDependant: true,
+              },
+              {
+                fullName: "Jane HOZZENFEUGEN",
+                dateOfBirth: "2006-06-01",
+                attendingPostSecondarySchool: "yes",
+                declaredOnTaxes: "yes",
+                validDependent: 17,
+                isavalidDependant: true,
+                pdDependentUpload: [
+                  {
+                    storage: "url",
+                    originalName: "dependants.txt",
+                    name: "dependants-d1966d14-93d5-41e7-815d-3fa97f3f8111.txt",
+                    url: "student/files/dependants-d1966d14-93d5-41e7-815d-3fa97f3f8111.txt",
+                    size: 0,
+                    type: "text/plain",
+                    hash: "1cb251ec0d568de6a929b520c4aed8d1",
+                  },
+                ],
+                dependantsIncomeTaxApplicationException: "",
+              },
+            ],
+            supportnocustodyDependants: "yes",
+            dependantCustodyFileUpload: [
+              {
+                storage: "url",
+                originalName: "dependants.txt",
+                name: "dependants-6d07f03d-1043-4bb5-9122-4a7b3476c6d6.txt",
+                url: "student/files/dependants-6d07f03d-1043-4bb5-9122-4a7b3476c6d6.txt",
+                size: 0,
+                type: "text/plain",
+                hash: "1cb251ec0d568de6a929b520c4aed8d1",
+              },
+            ],
+            dependantsSharedCustodyApplicationException: "",
+          },
+        },
+      ],
+    };
+
+    // Act
+    // Set application status to completed
+    application.applicationStatus = ApplicationStatus.Completed;
+    application.data = {
+      ...application.data,
+      studystartDate: "2024-02-01",
+      selectedOfferingDate: "2024-02-01",
+    };
+    await applicationRepo.save(application);
+
+    // Mock form.io dry-run submission
+    formService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: "studentDisabilityAppeal",
+      data: { data: payload.studentAppealRequests[0].formData },
+    });
+
+    // Assert
+    const response = await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.CREATED);
+
+    const newStudentAppealRequest = await studentAppealRequestRepo.findOne({
+      where: { studentAppeal: { id: response.body.id } },
+    });
+    const dependants = newStudentAppealRequest.submittedData.dependants;
+    expect(dependants.length).toBe(2);
```

Comment:
We can assert the whole input payload against the persisted one using strictEqual